src/main/java/com/hierynomus/smbj/session/Session.java

```diff
@@ -61,7 +63,7 @@
     private SMBEventBus bus;
     private final PathResolver pathResolver;
     private TreeConnectTable treeConnectTable = new TreeConnectTable();
-    private List<Session> nestedSessions = new ArrayList<>();
+    private Map<String, Session> nestedSessions = new HashMap<>();
```

Comment:
The change seems valid, however now there is potential race condition because there might be two threads trying to create the same nested session. One of these will be "lost". Please fix this case.